examples/ensemble/plot_random_forest_regression_criteria_comparison.py

```diff
@@ -62,24 +63,65 @@ def _test_forest(X, y, regr):
     return mean_squared_error(y, y_pred)
 
 
+def _prep_data(sim_dict, simulation_name, max_n_samples, n_dimensions, n_trials):
+    """Generate train and test data for all trials."""
+    # Get simulation parameters and validation dataset
+    sim, noise, (X_test, y_test) = simulations[simulation_name]
+    n_samples = int(max_n_samples)
+    n_dimensions = int(n_dimensions)
+
+    np.random.seed(random_state)
+    seeds = np.random.randint(1e8, size=n_trials)
+
+    sim_dict[simulation_name] = {}
+    for i in range(n_trials):
+        # Sample training data
+        if noise is not None:
+            X_train, y_train = sim(
+                n_samples=n_samples,
+                n_dimensions=n_dimensions,
+                noise=noise,
+                random_state=seeds[i],
+            )
+        else:
+            X_train, y_train = sim(
+                n_samples=n_samples, n_dimensions=n_dimensions, random_state=seeds[i]
+            )
+        sim_dict[simulation_name][i] = (
+            np.copy(X_train),
+            np.copy(y_train),
+            np.copy(X_test),
+            np.copy(y_test),
+        )
+    return sim_dict
+
+
 ###############################################################################
-def main(simulation_name, n_samples, criterion, n_dimensions, n_iter):
+def main(simulation_name, sim_data, n_samples, criterion, n_dimensions, n_iter):
     """Measure the performance of RandomForest under simulation conditions.
-
     Parameters
     ----------
     simulation_name : str
         Key from `simulations` dictionary.
+    sim_data: tuple
+        Contains X_train, y_train, X_test, and y_tests
+            X_train : np.array #TODO check this
+                All X training data for given simulation
+            y_train : np.array # TODO
+                All y training data for given simulation
+            X_test : np.array #TODO check this
+                All X testing data for given simulation
+            y_test : np.array # TODO
+                All y testing data for given simulation
     n_samples : int
         Number of training samples.
     criterion : string
         Split criterion used to train forest. Choose from
-        ("mse", "mae", "friedman_mse", "axis", "oblique").
+        ("mse", "mae", "friedman_mse").
```

Comment:
Again consider changing this to something more stylistic like is in `graspy` ([example](https://github.com/neurodata/graspy/blob/287bb762382aa00ae7d3d0702fcf53b71b989cc7/graspy/embed/omni.py#L79))